From 06d7cb39d71246cc1a0ef99167be0da6bc82ed01 Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Fri, 16 Feb 2024 17:38:22 +0800
Subject: [PATCH 47/87] mtk: mac80211: fix mac address to support hw path in
 station mode

Use AP's MLD address instead of using deflink.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 net/mac80211/iface.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/net/mac80211/iface.c b/net/mac80211/iface.c
index 0924afe9..82fa566b 100644
--- a/net/mac80211/iface.c
+++ b/net/mac80211/iface.c
@@ -950,7 +950,10 @@ static int ieee80211_netdev_fill_forward_path(struct net_device_path_ctx *ctx,
 			}
 		}
 
-		sta = sta_info_get(sdata, sdata->deflink.u.mgd.bssid);
+		if (ieee80211_vif_is_mld(&sdata->vif))
+			sta = sta_info_get(sdata, sdata->vif.cfg.ap_addr);
+		else
+			sta = sta_info_get(sdata, sdata->deflink.u.mgd.bssid);
 		break;
 	default:
 		goto out;
-- 
2.45.2

